rust/kcl-lib/src/std/sketch.rs

```diff
@@ -1778,6 +1781,680 @@ async fn inner_subtract_2d(
+    let coefficients = args.get_kw_arg_opt("coefficients", &RuntimeType::any_array(), exec_state)?;
+    let tag = args.get_kw_arg_opt("tag", &RuntimeType::tag_decl(), exec_state)?;
+
+    let new_sketch = inner_conic(
+        sketch,
+        start_tangent,
+        end,
+        end_tangent,
+        interior,
+        coefficients,
+        interior_absolute,
+        end_absolute,
+        tag,
+        exec_state,
+        args,
+    )
+    .await?;
+    Ok(KclValue::Sketch {
+        value: Box::new(new_sketch),
+    })
+}
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn inner_conic(
+    sketch: Sketch,
+    start_tangent: Option<[TyF64; 2]>,
+    end: Option<[TyF64; 2]>,
+    end_tangent: Option<[TyF64; 2]>,
+    interior: Option<[TyF64; 2]>,
+    coefficients: Option<[TyF64; 6]>,
+    interior_absolute: Option<[TyF64; 2]>,
+    end_absolute: Option<[TyF64; 2]>,
+    tag: Option<TagNode>,
+    exec_state: &mut ExecState,
+    args: Args,
+) -> Result<Sketch, KclError> {
+    let from: Point2d = sketch.current_pen_position()?;
+    let id = exec_state.next_uuid();
+
+    if (coefficients.is_some() && (start_tangent.is_some() || end_tangent.is_some()))
+        || (coefficients.is_none() && (start_tangent.is_none() && end_tangent.is_none()))
+    {
+        return Err(KclError::Type {
+            details: KclErrorDetails::new(
+                "Invalid combination of arguments. Either provide coefficients or interior".to_owned(),
+                vec![args.source_range],
+            ),
+        });
+    }
+
+    let (interior, end, relative) = match (interior, end, interior_absolute, end_absolute) {
+        (Some(interior), Some(end), None, None) => (interior, end, true),
+        (None, None, Some(interior_absolute), Some(end_absolute)) => (interior_absolute, end_absolute, false),
+        _ => return Err(KclError::Type {
+            details: KclErrorDetails::new(
+                "Invalid combination of arguments. Either provide (end, interior) or (endAbsolute, interiorAbsolute)"
+                    .to_owned(),
+                vec![args.source_range],
+            ),
+        }),
+    };
+
+    let (end, _) = untype_array(end);
+    let (interior, _) = untype_point(interior);
+
+    let (start_tangent, end_tangent) = if let Some(coeffs) = coefficients {
+        let (coeffs, _) = untype_array(coeffs);
+        (conic_tangent(coeffs, [from.x, from.y]), conic_tangent(coeffs, end))
+    } else {
+        let start = if let Some(start_tangent) = start_tangent {
+            let (start, _) = untype_point(start_tangent);
+            start
+        } else {
+            let previous_point = sketch
+                .get_tangential_info_from_paths()
+                .tan_previous_point(from.ignore_units());
+            let from = from.ignore_units();
+            [from[0] - previous_point[0], from[1] - previous_point[1]]
+        };
+
+        let (end_tan, _) = untype_point(end_tangent.unwrap());
```

Comment:
Even if we checked the proper combinations above, we shouldn't `unwrap()` since it panics. I'd rather we have a runtime error message that's maybe unreachable so that if the code gets refactored, there's never a possibility of a panic.